Sorry I didn't think it was useful. Here are some of the failed tests:

```
――――――――――――――――――――――――――――――――――――――――――――――――――― ERROR at setup of test_event_rrule_hourly_off_last ―――――――――――――――――――――――――――――――――――――――――――――――――――

    @pytest.fixture(name="calendar")
    def mock_private_cal():
        """Mock a private calendar."""
>       _calendar = _mock_calendar("Private")

tests/components/caldav/test_calendar.py:311: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/components/caldav/test_calendar.py:356: in _mock_calendar
    events.append(Event(None, "%d.ics" % idx, event, None, str(idx)))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Event(Event: 0.ics), client = None, url = '0.ics'
data = 'BEGIN:VCALENDAR\nVERSION:2.0\nPRODID:-//E-Corp.//CalDAV Client//EN\nBEGIN:VEVENT\nUID:1\nDTSTAMP:20171125T000000Z\nDT...180000Z\nSUMMARY:This is a normal event\nLOCATION:Hamburg\nDESCRIPTION:Surprisingly rainy\nEND:VEVENT\nEND:VCALENDAR\n'
parent = None, id = '0', props = None

    def __init__(
        self, client=None, url=None, data=None, parent=None, id=None, props=None
    ):
        """
        CalendarObjectResource has an additional parameter for its constructor:
         * data = "...", vCal data for the event
        """
        super(CalendarObjectResource, self).__init__(
            client=client, url=url, parent=parent, id=id, props=props
        )
>       assert self.parent
E       AssertionError

../python-caldav/caldav/objects.py:1693: AssertionError
----------------------------------------------------------------- Captured stderr setup ------------------------------------------------------------------
DEBUG:asyncio:Using selector: EpollSelector
------------------------------------------------------------------- Captured log setup -------------------------------------------------------------------
2023-01-16 07:09:02.542 DEBUG    MainThread asyncio:selector_events.py:54 Using selector: EpollSelector
                                                                                                                                            93% █████████▍

―――――――――――――――――――――――――――――――――――――――――――――――――――― ERROR at setup of test_event_rrule_hourly_ended ―――――――――――――――――――――――――――――――――――――――――――――――――――――

    @pytest.fixture(name="calendar")
    def mock_private_cal():
        """Mock a private calendar."""
>       _calendar = _mock_calendar("Private")

tests/components/caldav/test_calendar.py:311: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/components/caldav/test_calendar.py:356: in _mock_calendar
    events.append(Event(None, "%d.ics" % idx, event, None, str(idx)))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Event(Event: 0.ics), client = None, url = '0.ics'
data = 'BEGIN:VCALENDAR\nVERSION:2.0\nPRODID:-//E-Corp.//CalDAV Client//EN\nBEGIN:VEVENT\nUID:1\nDTSTAMP:20171125T000000Z\nDT...180000Z\nSUMMARY:This is a normal event\nLOCATION:Hamburg\nDESCRIPTION:Surprisingly rainy\nEND:VEVENT\nEND:VCALENDAR\n'
parent = None, id = '0', props = None

    def __init__(
        self, client=None, url=None, data=None, parent=None, id=None, props=None
    ):
        """
        CalendarObjectResource has an additional parameter for its constructor:
         * data = "...", vCal data for the event
        """
        super(CalendarObjectResource, self).__init__(
            client=client, url=url, parent=parent, id=id, props=props
        )
>       assert self.parent
E       AssertionError

../python-caldav/caldav/objects.py:1693: AssertionError
----------------------------------------------------------------- Captured stderr setup ------------------------------------------------------------------
DEBUG:asyncio:Using selector: EpollSelector
------------------------------------------------------------------- Captured log setup -------------------------------------------------------------------
2023-01-16 07:09:02.628 DEBUG    MainThread asyncio:selector_events.py:54 Using selector: EpollSelector
                                                                                                                                            95% █████████▌

――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― ERROR at setup of test_get_events ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

    @pytest.fixture(name="calendar")
    def mock_private_cal():
        """Mock a private calendar."""
>       _calendar = _mock_calendar("Private")

tests/components/caldav/test_calendar.py:311: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/components/caldav/test_calendar.py:356: in _mock_calendar
    events.append(Event(None, "%d.ics" % idx, event, None, str(idx)))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Event(Event: 0.ics), client = None, url = '0.ics'
data = 'BEGIN:VCALENDAR\nVERSION:2.0\nPRODID:-//E-Corp.//CalDAV Client//EN\nBEGIN:VEVENT\nUID:1\nDTSTAMP:20171125T000000Z\nDT...180000Z\nSUMMARY:This is a normal event\nLOCATION:Hamburg\nDESCRIPTION:Surprisingly rainy\nEND:VEVENT\nEND:VCALENDAR\n'
parent = None, id = '0', props = None

    def __init__(
        self, client=None, url=None, data=None, parent=None, id=None, props=None
    ):
        """
        CalendarObjectResource has an additional parameter for its constructor:
         * data = "...", vCal data for the event
        """
        super(CalendarObjectResource, self).__init__(
            client=client, url=url, parent=parent, id=id, props=props
        )
>       assert self.parent
E       AssertionError

../python-caldav/caldav/objects.py:1693: AssertionError
----------------------------------------------------------------- Captured stderr setup ------------------------------------------------------------------
DEBUG:asyncio:Using selector: EpollSelector
------------------------------------------------------------------- Captured log setup -------------------------------------------------------------------
2023-01-16 07:09:02.715 DEBUG    MainThread asyncio:selector_events.py:54 Using selector: EpollSelector
                                                                                                                                            98% █████████▊

――――――――――――――――――――――――――――――――――――――――――――――――――― ERROR at setup of test_get_events_custom_calendars ―――――――――――――――――――――――――――――――――――――――――――――――――――

    @pytest.fixture(name="calendar")
    def mock_private_cal():
        """Mock a private calendar."""
>       _calendar = _mock_calendar("Private")

tests/components/caldav/test_calendar.py:311: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/components/caldav/test_calendar.py:356: in _mock_calendar
    events.append(Event(None, "%d.ics" % idx, event, None, str(idx)))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Event(Event: 0.ics), client = None, url = '0.ics'
data = 'BEGIN:VCALENDAR\nVERSION:2.0\nPRODID:-//E-Corp.//CalDAV Client//EN\nBEGIN:VEVENT\nUID:1\nDTSTAMP:20171125T000000Z\nDT...180000Z\nSUMMARY:This is a normal event\nLOCATION:Hamburg\nDESCRIPTION:Surprisingly rainy\nEND:VEVENT\nEND:VCALENDAR\n'
parent = None, id = '0', props = None

    def __init__(
        self, client=None, url=None, data=None, parent=None, id=None, props=None
    ):
        """
        CalendarObjectResource has an additional parameter for its constructor:
         * data = "...", vCal data for the event
        """
        super(CalendarObjectResource, self).__init__(
            client=client, url=url, parent=parent, id=id, props=props
        )
>       assert self.parent
E       AssertionError

../python-caldav/caldav/objects.py:1693: AssertionError
```

I could paste them all but they really are pretty much identical.

--- a/homeassistant/components/caldav/calendar.py
+++ b/homeassistant/components/caldav/calendar.py
@@ -2,6 +2,7 @@
 from __future__ import annotations
 
 from datetime import date, datetime, timedelta
+from functools import partial
 import logging
 import re
 
@@ -172,7 +173,13 @@ class WebDavCalendarData:
         """Get all events in a specific time frame."""
         # Get event list from the current calendar
         vevent_list = await hass.async_add_executor_job(
-            self.calendar.date_search, start_date, end_date
+            partial(
+                self.calendar.search,
+                start=start_date,
+                end=end_date,
+                event=True,
+                expand=True,
+            )
         )
         event_list = []
         for event in vevent_list:
@@ -202,7 +209,12 @@ class WebDavCalendarData:
 
         # We have to retrieve the results for the whole day as the server
         # won't return events that have already started
-        results = self.calendar.date_search(start_of_today, start_of_tomorrow)
+        results = self.calendar.search(
+            start=start_of_today,
+            end=start_of_tomorrow,
+            event=True,
+            expand=True,
+        )
 
         # Create new events for each recurrence of an event that happens today.
         # For recurring events, some servers return the original event with recurrence rules

--- a/homeassistant/components/caldav/manifest.json
+++ b/homeassistant/components/caldav/manifest.json
@@ -2,7 +2,7 @@
   "domain": "caldav",
   "name": "CalDAV",
   "documentation": "https://www.home-assistant.io/integrations/caldav",
-  "requirements": ["caldav==0.9.1"],
+  "requirements": ["caldav==1.0.1"],
   "codeowners": [],
   "iot_class": "cloud_polling",
   "loggers": ["caldav", "vobject"]

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -507,7 +507,7 @@ btsmarthub_devicelist==0.2.3
 buienradar==1.0.5
 
 # homeassistant.components.caldav
-caldav==0.9.1
+caldav==1.0.1
 
 # homeassistant.components.circuit
 circuit-webhook==1.0.1

--- a/requirements_test_all.txt
+++ b/requirements_test_all.txt
@@ -408,7 +408,7 @@ bthome-ble==2.5.0
 buienradar==1.0.5
 
 # homeassistant.components.caldav
-caldav==0.9.1
+caldav==1.0.1
 
 # homeassistant.components.co2signal
 co2signal==0.4.2

--- a/tests/components/caldav/test_calendar.py
+++ b/tests/components/caldav/test_calendar.py
@@ -351,12 +351,12 @@ def _mocked_dav_client(*names, calendars=None):
 
 
 def _mock_calendar(name):
+    calendar = Mock()
     events = []
     for idx, event in enumerate(EVENTS):
-        events.append(Event(None, "%d.ics" % idx, event, None, str(idx)))
+        events.append(Event(None, "%d.ics" % idx, event, calendar, str(idx)))
 
-    calendar = Mock()
-    calendar.date_search = MagicMock(return_value=events)
+    calendar.search = MagicMock(return_value=events)
     calendar.name = name
     return calendar
 

